lib/app.js

```diff
@@ -642,17 +649,20 @@ module.exports = {
                                 'permission to use this Firebase');
           process.exit(1);
         }
+
         for (var i in defaultSettings) {
-          if (defaultSettings.hasOwnProperty(i) &&
+          if (i === 'public') {
+            if (settings.hasOwnProperty('public')) {
+              settings.public = path.resolve(defaultSettings.public, settings.public);
```

Comment:
We need to still make sure they don't specify a public directory that is above the directory that the `firebase.json` is in - I think that's still an important constraint to prevent people accidentally uploading their home folder etc.